stratum/hal/lib/bcm/bcm_chassis_manager.cc

```diff
@@ -1635,12 +1636,168 @@ bool BcmChassisManager::IsSingletonPortMatchesBcmPort(
 ::util::Status BcmChassisManager::WriteBcmConfigFile(
     const BcmChassisMap& base_bcm_chassis_map,
     const BcmChassisMap& target_bcm_chassis_map) const {
-  // TODO(unknown): Implement this function.
-  /*
   std::stringstream buffer;
-  RETURN_IF_ERROR(WriteStringToFile(buffer.str(), FLAGS_bcm_sdk_config_file));
-  */
-  return ::util::OkStatus();
+
+  // TODO(Yi): We use default node Id 0, need to support multiple nodes.
+  // PC_PM Table
+  YAML::Emitter pc_pm;
+  pc_pm << YAML::BeginDoc;
+  pc_pm << YAML::BeginMap;
+  pc_pm << YAML::Key << "device";
+  pc_pm << YAML::Value << YAML::BeginMap;
+  pc_pm << YAML::Key << "0";
+  pc_pm << YAML::Value << YAML::BeginMap;
+  pc_pm << YAML::Key << "PC_PM";
+  pc_pm << YAML::Value << YAML::BeginMap;
+
+  for (const auto& bcm_port : target_bcm_chassis_map.bcm_ports()) {
+    // Key is a map (PC_PM_ID: serdes_core)
+    pc_pm << YAML::Key << YAML::BeginMap << YAML::Key << "PC_PM_ID"
+          << YAML::Value << bcm_port.serdes_core() << YAML::EndMap;
+
+    pc_pm << YAML::Value << YAML::BeginMap;
+
+    pc_pm << YAML::Key << "PM_OPMODE" << YAML::Value << YAML::Flow
+          << YAML::BeginSeq << "PC_PM_OPMODE_DEFAULT" << YAML::EndSeq;
+
+    pc_pm << YAML::Key << "SPEED_MAX" << YAML::Value << YAML::Flow
```

Comment:
`SPEED_MAX` in the `PC_PM` table should not be set to the requested speed, as it prevents changing from 40G to 100G at runtime.